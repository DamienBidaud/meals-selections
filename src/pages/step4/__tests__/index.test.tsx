import { render, RenderResult, screen, waitFor } from "@testing-library/react";
import { useEffect } from "react";
import { atom, RecoilRoot, useRecoilValue } from "recoil";
import Step4 from "../index";
import { createBrowserRouter, RouterProvider } from "react-router-dom";

//@ts-ignore
export const RecoilObserver = ({ node, onChange }) => {
  const value = useRecoilValue(node);
  useEffect(() => onChange(value), [onChange, value]);
  return null;
};


const router = createBrowserRouter([
  {
    path: "/",
    element: <Step4 />,
  },
]);
const formState = atom({
  key: "form",
  default: {
    guests: 3,
      meal: "dinner",
      restaurant: "Mc Donalds",
      selectedMeals: [
        {
          meal: "Fries",
          serving: 3,
        },
      ],
  },
});

const onChange = jest.fn();
describe("Step 4", () => {
  let step4: RenderResult;
  beforeEach(() => {
    step4 = render(
      <RecoilRoot>
        <RouterProvider router={router} />
        <RecoilObserver node={formState} onChange={onChange} />
      </RecoilRoot>
    );
  });

  it("Should match the snapshot", () => {
    const { asFragment } = step4;
    expect(asFragment()).toMatchSnapshot();
  });
});
