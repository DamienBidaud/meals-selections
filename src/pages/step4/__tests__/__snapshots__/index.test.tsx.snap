// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step 4 Should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Review
    </h1>
    <dl>
      <dt>
        Meal
      </dt>
      <dd>
        dinner
      </dd>
      <dt>
        Guest(s)
      </dt>
      <dd>
        3
      </dd>
      <dt>
        Restaurant
      </dt>
      <dd>
        Mc Donalds
      </dd>
      <dt>
        Fries
      </dt>
      <dd>
        3
      </dd>
    </dl>
    <button
      class="sc-gswNZR bzVgbG"
    >
      Back
    </button>
  </div>
</DocumentFragment>
`;
